gcc/rust/typecheck/rust-hir-type-check-expr.cc

```diff
@@ -1065,11 +1065,16 @@ TypeCheckExpr::visit (HIR::MethodCallExpr &expr)
   if (candidates.size () > 1)
     {
       rich_location r (line_table, expr.get_method_name ().get_locus ());
+      std::string msg = "duplicate definitions for "
+			+ expr.get_method_name ().get_segment ().as_string ();
       for (auto &c : candidates)
-	r.add_range (c.candidate.locus);
+	{
+	  r.add_fixit_replace (c.candidate.locus, msg.c_str ());
+	  r.add_range (c.candidate.locus);
+	}
 
       rust_error_at (
-	r, "multiple candidates found for method %<%s%>",
+	r, ErrorCode::E0592, "duplicate definitions with name %<%s%>",
```

Comment:
hm, not sure which message we should keep here and which error code we should use

Reply:
I think we should keep the original error message, as this one could confuse the user since the error will appear on the use site rather than the definition site

Reply:
Done :)